python/cog/server/clients.py

```diff
@@ -213,7 +218,7 @@ async def upload_file(
 
         # this code path happens when running outside replicate without upload-url
         # in that case we need to return data uris
-        if url is None:
+        if url is None and not isinstance(fh, OutputURL):
             return file_to_data_uri(fh, content_type)
```

Comment:
I think this currently breaks the local development flow, it'll try and upload the `OutputURL` without a `url`... We need to return the URL as-is I think.

```suggestion
        if url is None:
            if isinstance(fh, OutputURL):
              return fh.url
            return file_to_data_uri(fh, content_type)
```